### PR TITLE
feat: reusable list item

### PR DESCRIPTION
**Pull request by winniezheng**

### CODE REVIEW

**Comment by winniezheng** [comment]
Example use:
const list = [
  {
    name: "Amy Farha",
    avatar_url:
      "https://s3.amazonaws.com/uifaces/faces/twitter/ladylexy/128.jpg",
    subtitle: "Vice President",
  },
  {
    name: "Chris Jackson",
    avatar_url:
      "https://s3.amazonaws.com/uifaces/faces/twitter/adhamdannaway/128.jpg",
    subtitle: "Vice Chairman",
  },
];

  <View>
    {list.map((l, i) => (
      <ListItem
        id={`${l.name}-${i}`}
        key={`${l.name}-${i}`}
        title={l.name}
        otherInformation={l.subtitle}
        showNav={true}
        iconName="md-basket"
      ></ListItem>
    ))}
  </View>

File: src/components/ui/ListItem/index.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { colors } from "@app/utils";
+
+interface ListItemProps extends RNListItemProps {
+  id: string;
+  title: string;
+  otherInformation: ReactNode | string; //for other elements to display below the title e.g. description and pending bar
+  iconName?: string;
+  iconColor?: string;
+  iconFamily?: string;
+  showNav: boolean; //show chevron right navigation icon
```
Comment (chrisrabe) [inline src/components/ui/ListItem/index.tsx]: maybe we could add a default value to have this as true by default since most list item has it

**Review by chrisrabe — CHANGES_REQUESTED** [review]
Just need default value for nav since most list item has that right chevron

File: src/components/ui/ListItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  onClick?: () => void;
+}
+
+const ListItem: React.FC<ListItemProps> = (props) => {
```
Comment (chrisrabe) [inline src/components/ui/ListItem/index.tsx:20]: It's better to do it like this for default values. Also, make the ones with default values as optional props

```
ListItem.defaultProps = {
  iconName: "md-basket",
  showNav: true,
  iconColor: "grey",
  iconFamily: "ionicon"
}
```

**Review by chrisrabe — CHANGES_REQUESTED** [review]
use defaultProps property instead

https://blog.bitsrc.io/understanding-react-default-props-5c50401ed37d

**Review by chrisrabe — APPROVED** [review]
looks good

**Pull request merged**